### DISCUSSION TITLE
How to log sql execution with R2dbc

### DISCUSSION THREAD

**Discussion by hantsy** [jOOQ/jOOQ#14498] [Q&A] [answered]
I follow the official post [Debug logging SQL with jOOQ](https://blog.jooq.org/debug-logging-sql-with-jooq/) and [this question on stackoverflow](https://stackoverflow.com/questions/11026214/how-can-one-see-the-sql-statements-that-jooq-executes), add the following config to Spring Boot project `application.properties`.

```
logging.level.org.jooq.tools.LoggerListener=DEBUG
``` 

But there is no any SQL statement in the log produced by jOOQ.

The above config may resolve the issue when using jOOQ with Jdbc, but my project is based on Spring Boot/Spring Data R2dbc/jOOQ.

**Comment by lukaseder** [comment, marked as answer]
The `LoggerListener` is an `ExecuteListener`, and thus, its lifecycle isn't yet supported in `R2DBC`, see https://github.com/jOOQ/jOOQ/issues/12221

You can probably implement an R2DBC `Connection` proxy relatively easily, which implements logging for you. Having said so, this is something that could be generally useful, so I'll add a `LoggingConnection` to jOOQ, just like we have for JDBC: https://github.com/jOOQ/jOOQ/issues/14499
